opm/parser/eclipse/Parser/MessageContainer.hpp

```diff
@@ -20,17 +20,27 @@
 #ifndef MESSAGECONTAINER_H
 #define MESSAGECONTAINER_H
 
-#endif // OPM_MESSAGECONTAINER_HEADER_INCLUDED
-
 #include <string>
 #include <vector>
 #include <memory>
 
 namespace Opm {
 
-    namespace MessageType {
+    struct Location {
+        Location() = default;
+        Location( const std::string& fn, int ln ) :
+            filename( fn ), lineno( ln ) {}
```

Comment:
This should throw for `ln <  0` - it should be freaking impossible to explicitly create a `Location` instance which evaluates to `false`.